### DISCUSSION TITLE
Tutorial, part 6 yields a broken project

### DISCUSSION THREAD

**Discussion by chgans** [encode/django-rest-framework#8564] [Potential Issue]
At the very end of part 6, https://www.django-rest-framework.org/tutorial/6-viewsets-and-routers/#using-routers
If you follow the tutorial, you end up with errors like
```
django.core.exceptions.ImproperlyConfigured: Could not resolve URL for hyperlinked relationship using view name "snippet-detail". You may have failed to include the related model in your API, or incorrectly configured the `lookup_field` attribute on this field.
```

As a beginner, it is not easy to understand what the problem is.

Turns out that `basename` needs to be removed.
from
```
router.register(r'snippets', views.SnippetViewSet,basename="snippets")
router.register(r'users', views.UserViewSet,basename="users")
```
to
```
router.register(r'snippets', views.SnippetViewSet)
router.register(r'users', views.UserViewSet)
```

This fixes the issue.
BTW, there's no explanation on why `basename` was needed.
I found the tutorial to be excellent! But IMO, the last part doesn't seem polished.

**Comment by ColemanDunn** [comment]
Yup was wondering the same thing. You can also just remove the plural ‘s’ in the base names as well. But yes very confusing and definitely an issue.
